portfolio/src/main/webapp/script.js

```diff
@@ -80,21 +79,60 @@ async function toggleCommentHistorySection() {
 }
 
 /**
- * Fetches and adds a history of comments, and the theoretical maximum and
+ * Comments include (1) comment submission; (2) comment history.
+ * * Authenticates the current user.
+ * * Fetches and adds a history of comments, and the theoretical maximum and
  * default number of comments, the total number of pages, and the current
  * page ID, to the page.
- * Backup and restore comment submission/history sections as necessary
- * according to the disabling and enabling of the comment history section,
- * and whether the user is logged in.
- * Show a warning if the user input value of the number of comments to display
- * or page ID is invalid, or if the latest user input may be a XSS attack.
+ * * Hides/backups and displays/restores comment submission/history sections as
+ * necessary according to the disabling and enabling of the comment history
+ * section, and whether the user is logged in.
+ * * Shows a warning if the user input value of the number of comments to
+ * display or page ID is invalid, or if the latest user comment may be a XSS
+ * attack.
  */
 async function addComments(pageId) {
+  await getAuthentication();
+
+  // Check whether the comment submission and history sections are enabled,
+  // and backup & restore the sections as needed. If the comment history
```

Comment:
The first sentence in this comment is kind of unnecessary given that you essentially say the same thing in the method description. In general, your code should speak for itself and that's where good variable and function names come in

Reply:
Got it, thanks!